sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -117,11 +114,16 @@ AccessToken TokenCredentialImpl::GetToken(
         request = shouldRetry(statusCode, *response);
         if (request == nullptr)
         {
+          auto const bodyStream = response->ExtractBodyStream();
+          auto const bodyVec = bodyStream ? bodyStream->ReadToEnd(context) : response->GetBody();
+          auto const responseBody
+              = std::string(reinterpret_cast<char const*>(bodyVec.data()), bodyVec.size());
```

Comment:
reinterpret_cast? Is there a simpler cast?

Reply:
Unfortunately, no:
```cpp
#include <cstdint>

int main() {
    return nullptr != static_cast<char const*>(static_cast<std::uint8_t const*>(nullptr));
}
```

```
<source>(4): error C2440: 'static_cast': cannot convert from 'const uint8_t *' to 'const char *'
<source>(4): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast
```
```
<source>:4:23: error: static_cast from 'const std::uint8_t *' (aka 'const unsigned char *') to 'const char *' is not allowed
    return nullptr != static_cast<char const*>(static_cast<std::uint8_t const*>(nullptr));
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
```
<source>: In function 'int main()':
<source>:4:23: error: invalid 'static_cast' from type 'const uint8_t*' {aka 'const unsigned char*'} to type 'const char*'
    4 |     return nullptr != static_cast<char const*>(static_cast<std::uint8_t const*>(nullptr));
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

C-style case would work, but that is basically a `reinterpret_cast` anyway - yes, with simpler syntax, but I think we would hate it :)